plugins/janus_nosip.c

```diff
@@ -1798,107 +1874,86 @@ static int janus_nosip_allocate_local_ports(janus_nosip_session *session, gboole
 			close(session->media.pipefd[1]);
 			session->media.pipefd[1] = -1;
 		}
+
+		static unsigned session_count = 0;
+		static int fds[DEFAULT_RTP_RANGE_MAX - DEFAULT_RTP_RANGE_MIN + 1] = {0, };
+		if(session_count == 0) {
+			unsigned n_ports_locked = 0;
+			for(uint16_t port = rtp_range_min; port < rtp_range_max - 4; port++) {
+				int fd = socket(AF_INET, SOCK_DGRAM, 0);
+				if(fd == -1) {
+					// https://askubuntu.com/questions/162229/how-do-i-increase-the-open-files-limit-for-a-non-root-user
+					JANUS_LOG(LOG_ERR, "Error creating audio sockets... index = %d %s\n", port - DEFAULT_RTP_RANGE_MIN, strerror(errno));
+					continue;
+				}
+				if(janus_nosip_bind_socket(fd, port)) {
+					continue;
+				}
+
+				fds[port - DEFAULT_RTP_RANGE_MIN] = fd;
+				n_ports_locked++;
+			}
+			JANUS_LOG(LOG_ERR, "-------------------------------------------\n");
+			JANUS_LOG(LOG_ERR, "number ports locked %u of %u\n", n_ports_locked, rtp_range_max - rtp_range_min);
+			JANUS_LOG(LOG_ERR, "-------------------------------------------\n");
+		}
+		session_count++;
 	}
+	uint64_t t_start = janus_get_monotonic_time();
 	/* Start */
-	int attempts = 100;	/* FIXME Don't retry forever */
-	if(session->media.has_audio) {
+	if(session->media.has_audio && 
+		(session->media.local_audio_rtp_port == 0 || session->media.local_audio_rtcp_port == 0)) {
+		if(session->media.audio_rtp_fd != -1) {
+			JANUS_LOG(LOG_WARN, "Audio RTP unbind socket detected, closing ...\n");
+			close(session->media.audio_rtp_fd);
+			session->media.audio_rtp_fd = -1;
+		}
+		if(session->media.audio_rtcp_fd != -1) {
+			JANUS_LOG(LOG_WARN, "Audio RTCP unbind socket detected, closing ...\n");
```

Comment:
What are these two log lines for, and why are they LOG_WARN? This means they'll be VERY verbose, and might confuse users if they're not really warnings.

Reply:
By design we do not have any sockets unbind. Something went wrong if this happened, but we can work normally.